sources/packages/backend/libs/services/src/disbursement-schedule/disbursement-schedule-shared.service.ts

```diff
@@ -698,4 +725,532 @@ export class DisbursementScheduleSharedService extends RecordDataModelService<Di
       .getRawOne<{ sum?: number }>();
     return +(total?.sum ?? 0);
   }
+
+  /**
+   * Disbursement and application details of the given disbursement.
+   * @param disbursementScheduleId disbursement schedule id of COE.
+   * @param locationId location id.
+   * @returns disbursement and application summary.
+   */
+  async getDisbursementAndApplicationSummary(
```

Comment:
Apparently, all the new methods in this service are related to COE in some way.
In the past, we concentrated too many methods in the same service just because they were "related to the same table".
I would prefer having a service specific for COE operations where we can have all these methods. Does it make sense?